src/builder.cr

```diff
@@ -174,8 +174,24 @@ module Mint
       {runtime + compiled.to_s, type_checker.artifacts}
     end
 
+    def get_service_worker_utils(sw_utils_path)
+      if sw_utils_path
+        raise RuntimeFileNotFound, {
+          "path" => sw_utils_path,
+        } unless File.exists?(sw_utils_path)
+        File.read(sw_utils_path)
+      else
+        Assets.read("sw-utils.js")
+      end
+    end
+
     def terminal
       Render::Terminal::STDOUT
     end
+
+    def service_worker(artifacts, relative, optimize, sw_utils_path)
+      worker = ServiceWorker.new(artifacts, relative, optimize)
+      "#{worker}#{get_service_worker_utils(sw_utils_path)}"
```

Comment:
Service worker utils should come first, I guess?

```suggestion
      "#{get_service_worker_utils(sw_utils_path)}#{worker}"
```

Reply:
In _Javascript_ the `var` declarations are hoisted (this is the case of the worker code), so it is not necessary.
But I don't mind the change.

Reply:
This is not an issue since the `Mint` variable is used later on, and it's in the global scope so it will be found. I do like that it's before though.